### DISCUSSION TITLE
monaco-editor and vscode-autohotkey2-lsp

### DISCUSSION THREAD

**Discussion by RaptorX** [thqby/vscode-autohotkey2-lsp#652] [General]
Im not a ts developer but im playing with the idea of having a monaco-editor powered app that would allow for your extension to be displayed in a custom website.

due to the way how vscode and monaco works this is not as simple as one might think but in monaco's faq they mention:

> I've written an extension for VS Code, will it work on the Monaco Editor in a browser? 
>

> No.

>
> **Note:** *If the extension is fully based on the [LSP](https://microsoft.github.io/language-server-protocol/) and if the language server is authored in JavaScript, then it would be possible.*

while playing wround i found out that i would need to load a few things like:

• monaco-editor 
• @codingame/monaco-vscode-api 
• monaco-languageclient 
• vscode-ws-jsonrpc 
• express

and then make all work on a server that uses Node.js and so on.

Do you have any idea of how to make all that work? have you ever played with monaco-editor at all besides vscode itself?

**Comment by thqby** [comment]
I haven't used it, but you can try installing the extension on monaco and see if it works. The extension works on vscode-web, but some features are limited.
